environs/config/config.go

```diff
@@ -595,6 +601,23 @@ func Validate(cfg, old *Config) error {
 			return fmt.Errorf("Invalid value for container-networking-method - %v", v)
 		}
 	}
+
+	if v, ok := cfg.defined[CloudInitUserDataKey].(string); ok && v != "" {
+		userDataMap, err := cfg.CloudInitUserData()
+		if err != nil {
+			return err
```

Comment:
And when we do the unmarshall, we either Annotate() the error or Trace()

Reply:
updated